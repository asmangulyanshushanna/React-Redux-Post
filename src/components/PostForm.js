import React from 'react';
import {connect} from 'react-redux';
import {addPost} from '../actions/postAction'

export function PostForm(props) {
 
 
  const onSubmit = (e) => {
      console.log('in submit')
      e.preventDefault();
      const title = e.target.elements.title.value
      const body = e.target.elements.body.value
      const postItem = {
          title,
          body
      }
      props.addPost(postItem);
      document.getElementById('form').reset();
  }
    return(
        <div>
            <h1>Add Post</h1>
            <form id='form' onSubmit={onSubmit}>
                <input type="text" name="title" /><br/>
                <textarea type="text" name="body" /><br/>
                <button>Submit</button>
            </form>
        </div>
        
    )
  }

//   const mapStateToProps = (state) => ({
//     posts: state.posts.items
//   })
  
  export default connect(null, {addPost})(PostForm);