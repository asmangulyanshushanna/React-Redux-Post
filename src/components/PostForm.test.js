import React from 'react';
import { shallow, configure } from 'enzyme';
import { PostForm } from './PostForm';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()})

describe('posts form component', () => {
    const addPost = jest.fn();
    const props = {
        addPost
    }
    const wrapper = shallow(<PostForm {...props} />)
    
    it('should render and match a stapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });
    it('should contain title', () => {
        const wrapper = shallow( <PostForm {...props} />).shallow();
        wrapper.instance().onSubmit = jest.fn();
        const {onSubmit} = wrapper.instance();
        const mockedEvent = {target: {}};
        wrapper.find('form').simulate('submit', mockedEvent);
        expect(onSubmit).toHaveBeenCalled();
    });
})