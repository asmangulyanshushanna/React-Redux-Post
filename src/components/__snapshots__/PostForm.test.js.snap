// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts form component should render and match a stapshot 1`] = `ShallowWrapper {}`;
